identity-enabler/holder-mobile-app/src/lib/scan.ts

```diff
@@ -0,0 +1,56 @@
+import { navigate } from "svelte-routing";
+import { wait } from "./helpers";
+import { loadingScreen } from "./store";
+import { playAudio } from "./ui/helpers";
+
+// We delay playing the valid or invalid sound in order not to overlap
+// with the scanning sound
+const PLAY_DELAY = 400;
+
+/**
+ * Handles data string captured by Camera, DataWedge or Image selection.
+ *
+ * @param decodedText The decoded string to verify and persist.
+ * @param method How the code was decoded.
+ * @returns Promise.
+ */
+export async function handleScannerData(decodedText: string, method: "Camera" | "File" | "DataWedge"): Promise<void> {
+    loadingScreen.set("Validating Claims...");
+    let scanSoundStart: number;
+
+    if (method !== "DataWedge") {
+        await playAudio("scanned");
+        scanSoundStart = Date.now();
+    }
+
+    let credentialSubject;
+
+    try {
+        credentialSubject = JSON.parse(decodedText);
+    } catch (err) {
+        console.error(err);
+        await handleError("Invalid JSON", scanSoundStart);
+        return;
+    }
```

Comment:
you need to check that credentialSubject is an object first of all, it could be a string ... because something like ```JSON.parse("\"hello\"")``` will parse into a string ... :) so you cannot make any assumptions on the decodedText